kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/VectorSpace.kt

```diff
@@ -0,0 +1,47 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.Dimension
+import scientifik.kmath.linear.Point
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.Space
+
+interface Vector<T, D : Dimension> : Point<T>
+
+interface VectorSpace<T, D : Dimension, V : Vector<T, D>> : Space<V> {
+    fun vectorFrom(x: Sequence<T>): V
+
+    // TODO what about it?
+    fun multiply(a: V, k: T): V = multiply(a, k as Number)
+
+    operator fun V.times(k: T) = multiply(this, k)
+    operator fun T.times(v: V) = multiply(v, this)
+
+    val dim: D
+    val field: ExtendedField<T>
```

Comment:
Why `ExtendedField`?

Reply:
To have a square root function implementation for T.

Reply:
You may create a type intersection to have `sqrt`, i.e. 

```kotlin
interface NeedSqrt<T, A> where A : PowerOperatrions<T>, A : Field<T> {
  val field: A
  
  fun useSqrt() = field { sqrt(one) }
}
```

Reply:
that way we will also parametrize `VectorSpace` with Field, I think it is not bad at all, since vector space is defined over some field and probably it makes sense to include this Field into vector space parametrization